File: src/solar/tool/CoolDown.ts
import BaseUtil from "solar/util/BaseUtil";

/**
 * 冷却时间
 */
export default class CoolDown {
    cool: number;
    lastWorkTime: number = -1000000;


    /**
     *
     * @param cool = 冷却时间
     */
    constructor(cool: number) {
        this.cool = cool;
    }


    /**
     * 判断是否已冷却完毕
     * @param doWork 如果为true 则冷却完毕时马上再次进入冷却状态
     */
    isReady(doWork = true): boolean {
        let r = (BaseUtil.getGameTime() - this.lastWorkTime) > this.cool
        if (r && doWork) {
            this.lastWorkTime = BaseUtil.getGameTime();
        }
        return r;
    }

    /**
     * 判断是否已冷却完毕
     * 如果冷却完毕则执行回调
     */
    ifReady(callBack: () => void): void {
        if (this.isReady()) {
            callBack();
        }
    }


}

File: scripts/solar/tool/CoolDown.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 1,["7"] = 1,["8"] = 6,["9"] = 6,["10"] = 6,["11"] = 15,["12"] = 8,["13"] = 16,["14"] = 15,["15"] = 24,["16"] = 24,["17"] = 24,["19"] = 25,["20"] = 26,["21"] = 27,["23"] = 29,["24"] = 24,["25"] = 36,["26"] = 37,["27"] = 38,["29"] = 36,["30"] = 6});
local ____exports = {}
local ____BaseUtil = require("solar.util.BaseUtil")
local BaseUtil = ____BaseUtil.default
____exports.default = __TS__Class()
local CoolDown = ____exports.default
CoolDown.name = "CoolDown"
function CoolDown.prototype.____constructor(self, cool)
    self.lastWorkTime = -1000000
    self.cool = cool
end
function CoolDown.prototype.isReady(self, doWork)
    if doWork == nil then
        doWork = true
    end
    local r = BaseUtil.getGameTime() - self.lastWorkTime > self.cool
    if r and doWork then
        self.lastWorkTime = BaseUtil.getGameTime()
    end
    return r
end
function CoolDown.prototype.ifReady(self, callBack)
    if self:isReady() then
        callBack(nil)
    end
end
____exports.default = CoolDown
return ____exports
